### PR TITLE
Add JS TypedArrays compat data

### PR DESCRIPTION
**Pull request by Elchi3**

### CODE REVIEW

File: javascript/builtins/TypedArray.json [hunk 1]
```diff
@@ -0,0 +1,2350 @@
...
+              "edge": {
+                "version_added": false
+              },
+              "edge_mobile": {
+                "version_added": false
+              },
+              "firefox": {
+                "version_added": "16",
+                "version_removed": "34",
+                "notes": "Was available in Aurora and Nightly channels only."
```
Comment (teoli2003) [inline javascript/builtins/TypedArray.json]: I think we should use "flag": { type : "compile_flag" } or similar for these cases. Having 16 - 34 in the table seems odd when it has not been actually released.

File: javascript/builtins/TypedArray.json [hunk 2]
```diff
@@ -0,0 +1,2350 @@
...
+              },
+              "firefox_android": {
+                "version_added": "38",
+                "version_removed": "34",
+                "notes": "Was available in Aurora and Nightly channels only."
```
Comment (teoli2003) [inline javascript/builtins/TypedArray.json]: I think we should use "flag": { type : "compile_flag" } or similar for these cases. Having 16 - 34 in the table seems odd when it has not been actually released.

File: javascript/builtins/TypedArray.json (hunk 1 above)
Comment (teoli2003) [reply javascript/builtins/TypedArray.json]: The note can be kept to indicate where the flag was activated.

File: javascript/builtins/TypedArray.json (hunk 2 above)
Comment (teoli2003) [inline javascript/builtins/TypedArray.json]: The note can be kept to indicate where the flag was activated.

File: javascript/builtins/TypedArray.json [hunk 3]
```diff
@@ -0,0 +1,2350 @@
...
+                "version_added": "12"
+              },
+              "edge_mobile": {
+                "version_added": true
+              },
+              "firefox": {
+                "version_added": "37"
+              },
+              "firefox_android": {
+                "version_ad
```
Comment (teoli2003) [inline javascript/builtins/TypedArray.json]: I think this note is useless

Comment (teoli2003) [inline javascript/builtins/TypedArray.json]: The alternative names where also on firefox for android…

File: javascript/builtins/ArrayBuffer.json [hunk 4]
```diff
@@ -0,0 +1,438 @@
...
+                "version_added": false
+              },
+              "chrome": {
+                "version_added": false
+              },
+              "chrome_android": {
+                "version_added": false
+              },
+              "edge": {
+                "version_added": false
```
[the remark this answers is not in the document]
Comment (Elchi3) [reply javascript/builtins/ArrayBuffer.json]: I think false is correct. It is just an experiment for now, not on any standard track.

File: javascript/builtins/ArrayBuffer.json [hunk 5]
```diff
@@ -0,0 +1,438 @@
...
+              },
+              "edge_mobile": {
+                "version_added": false
```
[the remark this answers is not in the document]
Comment (Elchi3) [reply javascript/builtins/ArrayBuffer.json]: dito

File: javascript/builtins/DataView.json [hunk 6]
```diff
@@ -0,0 +1,1302 @@
...
+              }
+            },
+            "status": {
+              "experimental": false,
+              "standard_track": true,
+              "deprecated": false
+            }
+          }
+        },
+        "getInt16": {
```
[the remark this answers is not in the document]
Comment (Elchi3) [reply javascript/builtins/DataView.json]: It's not, if you sort alphabetically, the order is getInt16, getInt32, getInt8

File: javascript/builtins/TypedArray.json [hunk 7]
```diff
@@ -0,0 +1,2350 @@
...
+              }
+            },
+            "status": {
+              "experimental": false,
+              "standard_track": true,
+              "deprecated": false
+            }
+          }
+        },
+        "length": {
```
[the remark this answers is not in the document]
Comment (Elchi3) [reply javascript/builtins/TypedArray.json]: It has an MDN page, I think array length is a bit special.

File: javascript/builtins/TypedArray.json (hunk 1 above)
Comment (Elchi3) [reply javascript/builtins/TypedArray.json]: I would keep this as a note until https://github.com/mdn/browser-compat-data/issues/338 is resoled.

File: javascript/builtins/TypedArray.json (hunk 2 above)
Comment (Elchi3) [reply javascript/builtins/TypedArray.json]: Same here, see https://github.com/mdn/browser-compat-data/issues/338

**Comment by Elchi3** [comment]
Updated. I've put removing ".0" and the sorting in separate commits, so you can re-review more easily.

File: javascript/builtins/DataView.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (teoli2003) [reply javascript/builtins/DataView.json]: My bad :-)

File: javascript/builtins/TypedArray.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (teoli2003) [reply javascript/builtins/TypedArray.json]: Sounds ok then.

**Comment by teoli2003** [comment]
LGTM, r!

**Comment by Elchi3** [comment]
Thanks!

**Pull request merged**